src/app/settings/sponsoring-org-row.component.ts

```diff
@@ -60,4 +78,56 @@ export class SponsoringOrgRowComponent {
     this.platformUtilsService.showToast("success", null, this.i18nService.t("reclaimedFreePlan"));
     this.sponsorshipRemoved.emit();
   }
+
+  private setStatus(
+    selfHosted: boolean,
+    toDelete?: boolean,
+    validUntil?: Date,
+    lastSyncDate?: Date
+  ) {
+    /*
+     * Possible Statuses:
+     * Requested (self-hosted only)
+     * Sent
+     * Active
+     * RequestRevoke
+     * RevokeWhenExpired
+     */
+
+    if (toDelete && validUntil) {
+      // They want to delete but there is a valid until date which means there is an active sponsorship
+      this.statusMessage = this.i18nService.t(
+        "revokeWhenExpired",
+        formatDate(validUntil, "MM/dd/yyyy", this.i18nService.locale)
+      );
+      this.statusClass = "text-danger";
+    } else if (toDelete) {
+      // They want to delete and we don't have a valid until date so we can
+      // this should only happen on a self-hosted install
+      this.statusMessage = this.i18nService.t("requestRemoved");
+      this.statusClass = "text-danger";
+    } else if (validUntil) {
+      // They don't want to delete and they have a valid until date
+      // that means they are actively sponsoring someone
+      this.statusMessage = this.i18nService.t("active");
+      this.statusClass = "text-success";
+    } else if (selfHosted && lastSyncDate) {
+      // We are on a self-hosted install and it has been synced but we have not gotten
+      // a valid until date so we can't know if they are actively sponsoring someone
+
+      // QUESTION: This has the same status as the cloud version, do we want to show something
+      // different and show next sync time?
```

Comment:
I don't totally know. Where have we landed on how often we let people sync? Is it hard coded to once a week because if so no we can't know but if we let them decide we can add an endpoint to view what the current orgs setting is and do some math.

Reply:
I think we don't show it for now and leave that to a future optimization if we want.